frontend/src/app/App.test.tsx

```diff
@@ -917,6 +917,51 @@ describe("App.onHistoryChange", () => {
       expect(instance.onPageChange).toHaveBeenLastCalledWith("sub_hash")
     })
   })
+
+  it("doesn't rerun when we are on the same page and the url contains an anchor", () => {
+    const pushStateSpy = jest.spyOn(window.history, "pushState")
+
+    const instance = wrapper.instance() as App
+    window.history.pushState({}, "", "#foo_bar")
+    jest.spyOn(instance, "onPageChange")
+    instance.onHistoryChange()
+
+    // check that onPageChange is not run when anchor is in the url and we are on the same page
```

Comment:
nit: should remove this comment -- I don't think it gives any more information that isn't already in the `it` description

Reply:
heads up this comment is still here